src/code/z_camera_data.c

```diff
@@ -59,3188 +85,4737 @@ char sCameraModeNames[][12] = {
+ */
+CameraModeValue sSetDirectedYawModeNormalData[] = {
+    { -10, CAM_DATA_Y_OFFSET },
+    { 280, CAM_DATA_EYE_DIST },
+    { 320, CAM_DATA_EYE_DIST_NEXT },
+    { -8, CAM_DATA_PITCH_TARGET },
+    { 20, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 10, CAM_DATA_XZ_UPDATE_RATE_TARGET },
+    { 80, CAM_DATA_MAX_YAW_UPDATE },
+    { 60, CAM_DATA_FOV },
+    { 80, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 0x0002, CAM_DATA_FLAGS },
+};
+
+/**
+ * Setting         Mode            Function
+ * ------------    ------------    --------
+ * DIRECTED_YAW    FOLLOWTARGET    KEEP1
+ */
+CameraModeValue sSetDirectedYawModeFollowTargetData[] = {
+    { -20, CAM_DATA_Y_OFFSET },
+    { 180, CAM_DATA_EYE_DIST },
+    { 200, CAM_DATA_EYE_DIST_NEXT },
+    { 35, CAM_DATA_SWING_YAW_INIT },
+    { 45, CAM_DATA_SWING_YAW_FINAL },
+    { 0, CAM_DATA_SWING_PITCH_INIT },
+    { -5, CAM_DATA_SWING_PITCH_FINAL },
+    { 20, CAM_DATA_SWING_PITCH_ADJ },
+    { 50, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 0x2001, CAM_DATA_FLAGS },
+    { -50, CAM_DATA_GROUND_Y_OFFSET },
+    { 30, CAM_DATA_GROUND_AT_LERP_STEP_SCALE },
+};
+
+/**
+ * Setting         Mode    Function
+ * ------------    ----    --------
+ * DIRECTED_YAW    TALK    KEEP3
+ */
+CameraModeValue sSetDirectedYawModeTalkData[] = {
+    { -80, CAM_DATA_Y_OFFSET },
+    { 200, CAM_DATA_EYE_DIST },
+    { 250, CAM_DATA_EYE_DIST_NEXT },
+    { 30, CAM_DATA_SWING_YAW_INIT },
+    { 10, CAM_DATA_SWING_YAW_FINAL },
+    { -8, CAM_DATA_SWING_PITCH_INIT },
+    { -8, CAM_DATA_SWING_PITCH_FINAL },
+    { 30, CAM_DATA_SWING_PITCH_ADJ },
+    { 50, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 10, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 0x3520, CAM_DATA_FLAGS },
+};
+
+/**
+ *=====================================================================
+ *                   Custom Data: NORMAL4 Setting
+ *=====================================================================
+ */
+
+/**
+ * Setting    Mode    Function
+ * -------    ----    --------
+ * NORMAL4    TALK    KEEP3
+ */
+CameraModeValue sNormal4ModeTalkData[] = {
+    { -30, CAM_DATA_Y_OFFSET },
+    { 70, CAM_DATA_EYE_DIST },
+    { 200, CAM_DATA_EYE_DIST_NEXT },
+    { 40, CAM_DATA_SWING_YAW_INIT },
+    { 10, CAM_DATA_SWING_YAW_FINAL },
+    { 0, CAM_DATA_SWING_PITCH_INIT },
+    { 5, CAM_DATA_SWING_PITCH_FINAL },
+    { 70, CAM_DATA_SWING_PITCH_ADJ },
+    { 45, CAM_DATA_FOV },
+    { 50, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 10, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 0x35A0, CAM_DATA_FLAGS },
+};
+
+/**
+ * =====================================================================
+ *    CAMERA SETTINGS: USAGE OF FUNCTIONS AND DATA FOR SPECIFIC MODES
+ * =====================================================================
+ */
+
+/**
+ * Camera Setting: NORMAL0
+ */
+CameraMode sCamSetNormal0Modes[] = {
+    { CAM_FUNC_NORM1, ARRAY_COUNT(sSetNormal0ModeNormalData), sSetNormal0ModeNormalData }, // CAM_MODE_NORMAL
```

Comment:
If all these entries are always the array count + pointer, then I think a macro should be used to avoid duplicating the array reference

Reply:
It's pretty messy as is yeah so macros would be really nice. How would you suggest making the macros?

Reply:
I didn't have anything specific in mind, I was thinking just a basic macro with 2 arguments which would generate an entry.
Something like this I guess (possibly with arguments swapped if it makes more sense):
```c
#define CAM_MODE(func, data) { func, ARRAY_COUNT(data), data }